### PR TITLE
[Fluent][Ready For Review] VM extension images & Sample restructuring

### PR DESCRIPTION
**Pull request by anuchandy**
1. Support for VM extension images
2. Restructuring sample so that each one o them can be executed independendly 
3. Adding VM extension sample
4. Ensuring root resource gets updated on Apply()

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anuchandy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anuchan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/ComputeRoles.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 using System.Linq;
 using System.Threading.Tasks;
 
-namespace Microsoft.Azure.Management.V2.Compute
+namespace Microsoft.Azure.Management.Compute.Models
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/ComputeRoles.cs]: you should have Fluent in the namespace. We do not want this libraries to collide with AutoRest generated SDK's

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/VirtualMachineExtensionImageVersionsImpl.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT License. See License.txt in the project root for
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/VirtualMachineExtensionImageVersionsImpl.cs]: please change this to C# style license header.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageImpl.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        /// <returns>the region in which virtual machine extension image is available</returns>
+        string Microsoft.Azure.Management.V2.Compute.IVirtualMachineExtensionImage.RegionName
+        {
+            get
+            {
+                return this.RegionName as string;
+            }
+        }
+        /// <returns>the schema defined by publisher, where extension consumers should provide settings in a matching schema</returns>
+        /// <returns><p></returns>
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageImpl.cs]: >/// <returns><p></returns> [](start = 8, length = 26)

there should not be to <return> tags.

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageTypeImpl.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/**
+* Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageTypeImpl.cs]: please change to C# style license header

**Comment by hovsepm** [comment]
> using System;

Please add License header

---
Refers to: src/ResourceManagement/Resource/Microsoft.Azure.Management.V2.Resource/Core/EnumHelper.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

**Comment by hovsepm** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

please bring license header back 

---
Refers to: src/ResourceManagement/Azure/Samples/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = True)

File: src/ResourceManagement/Azure/Samples/ListVirtualMachineImages.cs [hunk 5]
```diff
@@ -1,22 +1,21 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (hovsepm) [inline src/ResourceManagement/Azure/Samples/ListVirtualMachineImages.cs]: bring license header back

**Comment by hovsepm** [comment]
> using System.Reflection;

please add license header 

---
Refers to: src/ResourceManagement.Samples/Compute.Samples/ListVirtualMachineImages/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

**Comment by hovsepm** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

license header

---
Refers to: src/ResourceManagement/Azure/Samples/ManageAvailabilitySet.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = True)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Compute.Samples/ManageAvailabilitySet/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

**Comment by hovsepm** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

License header

---
Refers to: src/ResourceManagement/Azure/Samples/ManageVirtualMachine.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = True)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Compute.Samples/ManageVirtualMachine/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

**Comment by hovsepm** [comment]
> using Microsoft.Azure.Management;

License header

---
Refers to: src/ResourceManagement.Samples/Compute.Samples/ManageVirtualMachineExtension/Program.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Compute.Samples/ManageVirtualMachineExtension/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

File: src/ResourceManagement/Azure/Samples/DeployUsingArmTemplate.cs [hunk 6]
```diff
@@ -1,18 +1,18 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (hovsepm) [inline src/ResourceManagement/Azure/Samples/DeployUsingArmTemplate.cs]: License header

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Resource.Samples/DeployUsingARMTemplate/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

File: src/ResourceManagement/Azure/Samples/ManageResource.cs [hunk 7]
```diff
@@ -1,15 +1,15 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (hovsepm) [inline src/ResourceManagement/Azure/Samples/ManageResource.cs]: License header

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Resource.Samples/ManageResource/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

**Comment by hovsepm** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

License header 

---
Refers to: src/ResourceManagement/Azure/Samples/ManageResourceGroup.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = True)

File: src/ResourceManagement.Samples/Resource.Samples/ManageResourceGroup/Properties/AssemblyInfo.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
```
Comment (hovsepm) [inline src/ResourceManagement.Samples/Resource.Samples/ManageResourceGroup/Properties/AssemblyInfo.cs]: License header

File: src/ResourceManagement/Azure/Samples/Utilities.cs [hunk 9]
```diff
@@ -1,6 +1,5 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (hovsepm) [inline src/ResourceManagement/Azure/Samples/Utilities.cs]: license header

File: src/ResourceManagement.Samples/Samples/Samples/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
```
Comment (hovsepm) [inline src/ResourceManagement.Samples/Samples/Samples/Properties/AssemblyInfo.cs]: License header

**Comment by hovsepm** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

License header

---
Refers to: src/ResourceManagement/Azure/Samples/ManageStorageAccount.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = True)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Storage.Samples/ManageStorageAccount/Properties/AssemblyInfo.cs:1 in 5fc66ba. [](commit_id = 5fc66ba61ff069d92116e585ee7bb117b919652f, deletion_comment = False)

File: global.json [hunk 11]
```diff
@@ -7,5 +7,11 @@
     "src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager",
     "src/ResourceManagement/Storage/Microsoft.Azure.Management.Storage",
     "src/ResourceManagement/Azure/Microsoft.Azure.Management",
+    "src/ResourceManagement/Storage",
```
Comment (hovsepm) [inline global.json]: >Storage [](start = 28, length = 7)

Please remove wring entries

**Comment by hovsepm** [comment]
@anuchandy looks like I broke you again :) Please pull from upstream and resolve merge conflicts.

**Comment by anuchandy** [comment]
@hovsepm thanks for review, i have addressed your feedback, please take a look

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageImpl.cs [hunk 12]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information
+
+namespace Microsoft.Azure.Management.V2.Compute
+{
+
+    using Microsoft.Azure.Management.V2.Resource.Core;
+    using Microsoft.Azure.Management.Compute.Models;
+    public partial class VirtualMachineExtensionImageImpl
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageImpl.cs]: >public [](start = 4, length = 6)

should be internal class

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImagesImpl.cs [hunk 13]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information
+
+namespace Microsoft.Azure.Management.V2.Compute
+{
+
+    using Microsoft.Azure.Management.V2.Resource.Core;
+    using Microsoft.Azure.Management.V2.Resource.Core.CollectionActions;
+    public partial class VirtualMachineExtensionImagesImpl
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImagesImpl.cs]: >public [](start = 4, length = 6)

should be internal class

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageTypeImpl.cs [hunk 14]
```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information. 
+
+namespace Microsoft.Azure.Management.V2.Compute
+{
+    using Microsoft.Azure.Management.Compute.Models ;
+    using Microsoft.Azure.Management.V2.Resource.Core;
+    public partial class VirtualMachineExtensionImageTypeImpl
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/Domain/InterfaceImpl/VirtualMachineExtensionImageTypeImpl.cs]: >public [](start = 4, length = 6)

should be internal

**Comment by hovsepm** [comment]
> using System.Reflection;

No License header

---
Refers to: src/ResourceManagement.Samples/Common/Common/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Compute/ListVirtualMachineExtensionImages/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header 

---
Refers to: src/ResourceManagement.Samples/Compute/ListVirtualMachineImages/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Compute/ManageAvailabilitySet/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Compute/ManageVirtualMachine/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header 

---
Refers to: src/ResourceManagement.Samples/Compute/ManageVirtualMachineExtension/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header 

---
Refers to: src/ResourceManagement.Samples/RedisCache/ManageRedis/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header 

---
Refers to: src/ResourceManagement.Samples/Resource/DeployUsingARMTemplate/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Resource/ManageResource/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Resource/ManageResourceGroup/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
> using System.Reflection;

License header

---
Refers to: src/ResourceManagement.Samples/Storage/ManageStorageAccount/Properties/AssemblyInfo.cs:1 in de8ddf2. [](commit_id = de8ddf22e459e098f0d88d457b0db657e652dc55, deletion_comment = False)

**Comment by hovsepm** [comment]
:clock1:

**Pull request merged**